influxql/ast.go

```diff
@@ -953,6 +953,31 @@ func (s *SelectStatement) HasWildcard() bool {
 	return false
 }
 
+// HasFieldWildcard returns whether or not the select statement has at least 1 wildcard in the fields
+func (s *SelectStatement) HasFieldWildcard() bool {
```

Comment:
Yeah, I haven't reviewed any of this code yet or refactored (hence the WIP in the title).  Makes sense.